### ISSUE TITLE
WebSocket/CDI integration is broken in Jetty 10

### ISSUE THREAD

**Issue by rdicroce** [eclipse/jetty.project#6106]
**Jetty version**
10.0.1

**Java version**
11.0.8

**OS type/version**
Windows 10 Pro 20H2

**Description**
Classes annotated with `@ServerEndpoint` do not receive CDI integration. You can reproduce the problem easily by upgrading https://github.com/jetty-project/embedded-jetty-weld to Jetty 10. The code needs a few tweaks since some things were moved around, but after that's fixed, you'll notice that TimeSocket blows up due to a NPE because the logger hasn't been injected. After digging around in the code a bit, the cause appears to be that nothing ever calls the ObjectFactory to decorate the actual endpoint class.

Workaround: specify a Configurator on the `@ServerEndpoint`. However, this does not work:
```java
	@Override
	public <T> T getEndpointInstance(Class<T> endpointClass) throws InstantiationException {
		return CDI.current().select(endpointClass).get();
	}
```
At least, it doesn't work if you're using Weld and the endpoint class is ApplicationScoped. In that case, Weld will return a proxy subclass, which creates a problem in JavaxWebSocketServerFrameHandlerFactory#getMetadata() because it looks for `@ServerEndpoint` but that annotation is not `@Inherited`. Therefore the annotation is not found and the configuration is ignored.

Instead, this works, if all you need is injection:
```java
	@Override
	public <T> T getEndpointInstance(Class<T> endpointClass) throws InstantiationException {
		return new Unmanaged<>(endpointClass).newInstance().produce().inject().get();
	}
```

**Comment by lachlan-roberts** [comment]
@rdicroce this issue is fixed by PR #6106, I have verified that https://github.com/jetty-project/embedded-jetty-weld now works if you run it against the current 10.0.x branch. This fix will be released in 10.0.3.

**Comment by rdicroce** [comment]
@lachlan-roberts Thanks for the fix!
